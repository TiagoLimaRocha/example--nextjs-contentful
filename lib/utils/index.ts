export { fromMarkdown } from './fromMarkdown';
export { match } from './match';
export { throttle } from './throttle';
export { renderComponents } from './renderComponents';
export { renderBgImage } from './renderBgImage';
export { renderParagraphs } from './renderParagraphs';
