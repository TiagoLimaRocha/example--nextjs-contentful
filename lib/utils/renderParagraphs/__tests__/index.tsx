import { render, screen } from '@testing-library/react';

import { renderParagraphs } from '@lib/utils';

import { dataMock } from '@mocks/utils/renderParagraphs/data.mock';

describe('renderParagraphs()', () => {
  it('should render multiple paragraphs', () => {
    render(<>{renderParagraphs({ body: dataMock })}</>);

    const paragraphs = dataMock.split('\n');

    paragraphs.forEach((paragraph) => {
      expect(screen.getByText(paragraph)).toBeInTheDocument();
    });
  });

  it('should render multiple paragraphs with a specific classname when provided', () => {
    const mockClassname = 'MOCK_CLASSNAME';
    const paragraphs = dataMock.split('\n');

    render(<>{renderParagraphs({ body: dataMock, className: mockClassname })}</>);

    paragraphs.forEach((paragraph) => {
      const snap = screen.getByText(paragraph);

      expect(snap).toBeInTheDocument();
      expect(snap).toHaveClass(mockClassname);
    });
  });

  it('should render parsed markup when the body is a markdown string', () => {
    const mockMarkdownString = `
      ## Formating

      **bold text**, 
      
      *italic text*, 
      
      <u>underlined text</u>, 
      
      ~~striked out~~,
      
      [link example](www.google.com)
      
      > quote example
      
      ## Lists
      
      ### Unordered
      
      - item 1
      - item 2
      - item 3
      
      ### Ordered
      
      1. Item 1
      2. Item 2
      3. Item 3
    `;

    const output = render(<>{renderParagraphs({ body: mockMarkdownString, hasMarkdown: true })}</>);

    expect(output).toMatchSnapshot();
  });

  it('should not render anything when body is not provided', () => {
    render(<>{renderParagraphs({ body: '' })});</>);

    const paragraphs = dataMock.split('\n');

    paragraphs.forEach((paragraph) => {
      const snap = screen.queryByText(paragraph);

      expect(snap).toBeNull();
    });
  });
});
