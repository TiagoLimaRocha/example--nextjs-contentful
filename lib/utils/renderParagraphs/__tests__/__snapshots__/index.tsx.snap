// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderParagraphs() should render parsed markup when the body is a markdown string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <pre>
        <code>
            ## Formating

  **bold text**, 
  
  *italic text*, 
  
  &lt;u&gt;underlined text&lt;/u&gt;, 
  
  ~~striked out~~,
  
  [link example](www.google.com)
  
  &gt; quote example
  
  ## Lists
  
  ### Unordered
  
  - item 1
  - item 2
  - item 3
  
  ### Ordered
  
  1. Item 1
  2. Item 2
  3. Item 3

        </code>
      </pre>
      

    </div>
  </body>,
  "container": <div>
    <pre>
      <code>
          ## Formating

  **bold text**, 
  
  *italic text*, 
  
  &lt;u&gt;underlined text&lt;/u&gt;, 
  
  ~~striked out~~,
  
  [link example](www.google.com)
  
  &gt; quote example
  
  ## Lists
  
  ### Unordered
  
  - item 1
  - item 2
  - item 3
  
  ### Ordered
  
  1. Item 1
  2. Item 2
  3. Item 3

      </code>
    </pre>
    

  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
